test/crawlers/cheerio_crawler.test.js

```diff
@@ -629,49 +638,51 @@ describe('CheerioCrawler', () => {
             await crawler.run();
             delete process.env[ENV_VARS.PROXY_PASSWORD];
 
+            utilsRequestMock.verify();
             // expect(proxies).to.have.lengthOf(1);
             expect(proxies[0]).toEqual(proxy);
             expect(proxies[1]).toEqual(proxy);
             expect(proxies[2]).toEqual(proxy);
             expect(proxies[3]).toEqual(proxy);
         });
 
-        test('should work with useApifyProxy and other opts', async () => {
+        test('should handleRequestFunction contains proxy object', async () => {
```

Comment:
We should also add a test that confirms the `proxy.sessionId` is correctly collected from the correct `session`, because this cannot be tested in `ProxyConfiguration` tests. (they can't check whether the Crawler actually calls the right functions and passes the right parameters)

Reply:
Created tests for that.